src/components/Ideas/CompIdeaDetail.js

```diff
@@ -0,0 +1,45 @@
+import React, { useContext, useEffect, useState } from "react"
+import { IdeaContext } from "./IdeaProvider"
+import { useParams, useHistory } from "react-router-dom"
+import "./Idea.css"
+
+export const CompIdeaDetail = (props) => {
+    const { ideas, getCompleteIdeas, deleteIdea } = useContext (IdeaContext)
+    const [ idea, setIdea ] = useState({ artist: {} })
+
+    const { ideaId } = useParams()
+
+    useEffect(() => {
+        if(props.idea){
+            setIdea(props.idea)
+        } else {
+            const thisIdea = idea.find(i => i.id === parseInt(ideaId)) || { artist: {} }
+            setIdea(thisIdea)
+        }
+    }, [idea, ideas])
+
+    useEffect(() => {getCompleteIdeas()}, [])
```

Comment:
I'm not sure if this is already doing this, but should this not look at user-specific data?